src/test/kotlin/me/spradling/gift/core/tests/unit/api/GetAccountsHandlerTests.kt

```diff
@@ -0,0 +1,118 @@
+package me.spradling.gift.core.tests.unit.api
+
+import io.netty.handler.codec.http.HttpResponseStatus
+import io.vertx.core.Future
+import me.spradling.gift.core.api.extensions.unwrap
+import me.spradling.gift.core.api.models.ApiRequest
+import me.spradling.gift.core.api.models.errors.ErrorDetails
+import me.spradling.gift.core.api.routes.v1.GetAccountsHandler
+import me.spradling.gift.core.database.memory.InMemoryGiftCommitStorageClient
+import me.spradling.gift.core.tests.unit.UnitTestBase
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.*
+import org.mockito.Mockito
+import org.mockito.Mockito.`when`
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@DisplayName("When I call `handleRequest` on GetAccountsHandler,")
+internal class GetAccountsHandlerTests : UnitTestBase() {
+
+  private val getHandler = GetAccountsHandler(inMemoryStorageClient)
+  private val mockInMemoryStorageClient = Mockito.mock(InMemoryGiftCommitStorageClient::class.java)
+
+  @Nested
+  @DisplayName("given no results")
+  inner class NoResults {
+
+    @Test
+    @DisplayName("request should fail with Not Found error")
+    fun failsWithNotFoundError() {
+      val response = getHandler.handleRequest(ApiRequest(mockContext)).unwrap()
+
+      verifyErrorResponse(response, ErrorDetails.RESOURCE_NOT_FOUND)
+    }
+  }
+
+  @Nested
+  @DisplayName("given a valid request")
+  inner class ValidRequest {
+
+    @BeforeEach
+    fun setup() {
+      inMemoryStorageClient.createAccount(storageConverter.convert(validAccounts["api"]!!))
```

Comment:
I don't think you are actually using this now. But now that I think about it, why didn't you just call this twice in order to start returning multiple entries?